clang/lib/Sema/SemaBounds.cpp

```diff
@@ -4685,11 +4800,20 @@ namespace {
     // CreateIntegerLiteral returns an integer literal with Ty type.
     // If Ty denotes a pointer to an integer type (char *, ptr<int>, etc.),
     // CreateIntegerLiteral returns an integer literal with Ty's pointee type.
+    // If Ty denotes a pointer to a non-integer type (float *, ptr<double>,
```

Comment:
I found this comment and code confusing.   Are you overloading creation of the integer literal to account for the different bitwidth of integer literals required for integer arithmetic vs. pointer arithmetic involving a literal.  For pointer arithmetic, if the pointee type is an integer type, why would you make the type of the integer literal match the pointee type?  An example, for "char *", it is valid to add an integer larger than a char to the pointer.